tgui/packages/tgui/interfaces/ChemiCompiler/ChemiCompilerTextArea.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Garash2k (https://github.com/Garash2k)
+ * @license ISC
+ */
+
+import { useBackend } from '../../backend';
+import { ChemiCompilerData } from './type';
+import { TextArea } from '../../components';
+
+export const ChemiCompilerTextArea = (_props, context) => {
+  const { act, data } = useBackend<ChemiCompilerData>(context);
+  const { inputValue, loadTimestamp, theme } = data;
+  return (
+    <TextArea
+      value={inputValue}
+      onInput={(_event, value) => { act('updateInputValue', { value }); }}
+      height="100%"
+      // The load button would break if we pressed it between the input's act and the next refresh.
+      // This ensures a refresh after every load button click
```

Comment:
The input value isn't stored locally so there's nothing to update. Ideally it would be stored locally using a local state, then it could be updated, but I couldn't figure out a way to make the load button not constantly override the input after a load click. It's hard to make the whole thing work without a single callback.

Reply:
~~Prompting with an idea, as I know it's moderately involved so I can help with the details if needed (please ping in coder channel Discord if you want to talk about it):~~

~~Could you use local state for the value, have a load button to "submit" that value, and then in the `ui_act` part of that include the timestamp as a `last_updated` field (alternatively, just increment a number if that doesn't give the right granularity). Include that in the `ui_data`, then use a diff between that value and what you last saw it as (in a local state) to determine if you need to update the text area's value or not?~~

I'm leaving the above in for posterity because funny, I wrote it out without really reading the code and riffing off of ZeWaka's idea instead of reading what you have. I think using a `loadTimestamp` is absolutely fine. I would be a little cautious of using it as the `key` as that involves unmounting/remounting and may lose focus, but would need to play with it a bit to see if we care about that.

Reply:
I don't think there's a focus loss issue. I'll only change when the user clicks the load button, which means that focus is already lost. Plus I'd say with the contents of the textarea changing, a loss of focus is desirable.